cli/dcoscli/job/main.py

```diff
@@ -0,0 +1,807 @@
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'update'],
+            arg_keys=['<job-id>', '<schedule-file>'],
+            function=_update_schedules),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'show'],
+            arg_keys=['<job-id>', '--json'],
+            function=_show_schedule),
+
+        cmds.Command(
+            hierarchy=['job', 'show', 'runs'],
+            arg_keys=['<job-id>', '<run-id>', '--q'],
+            function=_show_runs),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'remove'],
+            arg_keys=['<job-id>', '<schedule-id>'],
+            function=_remove_schedule),
+
+        cmds.Command(
+            hierarchy=['job', 'list'],
+            arg_keys=['--json'],
+            function=_list),
+
+        cmds.Command(
+            hierarchy=['job', 'history'],
+            arg_keys=['<job-id>','--json','--show-failures'],
+            function=_history),
+
+        cmds.Command(
+            hierarchy=['job', 'remove'],
+            arg_keys=['<job-id>','--stopCurrentJobRuns'],
+            function=_remove),
+
+        cmds.Command(
+            hierarchy=['job', 'add'],
+            arg_keys=['<job-file>'],
+            function=_add_job),
+
+        cmds.Command(
+            hierarchy=['job', 'update'],
+            arg_keys=['<job-file>'],
+            function=_update_job),
+
+        cmds.Command(
+            hierarchy=['job', 'show'],
+            arg_keys=['<job-id>'],
+            function=_show),
+
+        cmds.Command(
+            hierarchy=['job'],
+            arg_keys=['--config-schema', '--info'],
+            function=_job)
+    ]
+
+
+def _job(config_schema=False, info=False):
+    """
+    :param config_schema: Whether to output the config schema
+    :type config_schema: boolean
+    :param info: Whether to output a description of this subcommand
+    :type info: boolean
+    :returns: process return code
+    :rtype: int
+    """
+
+    if config_schema:
+        schema = _cli_config_schema()
+        emitter.publish(schema)
+    elif info:
+        _info()
+    else:
+        doc = default_command_info("job")
+        emitter.publish(options.make_generic_usage_message(doc))
+        return 1
+
+    return 0
+
+
+def _remove_schedule(job_id, schedule_id):
+    """
+    :param job_id: Id of the job
+    :type job_id: str
+    :param schedule_id: Id of the schedule
+    :type schedule_id: str
+    :returns: process return code
+    :rtype: int
+    """
+    response = None
+
+    try:
+     response = _do_request("{}/{}/schedules/{}".format(_get_api_url('v1/jobs'),
+        job_id, schedule_id), 'DELETE')
+    except DCOSHTTPException as e:
+        if e.response.status_code == 404:
+            emitter.publish("Schedule or job ID does NOT exist.")
+        return 1
+    except DCOSException as e:
+        emitter.publish("Unable to remove schedule ID '{}' for job ID '{}'".format(schedule_id, job_id))
+        return 1
```

Comment:
instead of emitting + returning 1 we should raise `DCOSException("error message here")`